fog/overseer/server/src/service.rs

```diff
@@ -118,4 +140,46 @@ where
 
         Ok(response)
     }
+
+    pub fn get_ingest_summaries(&self) -> Result<GetIngestSummariesResponse, String> {
+        let mut ingest_summaries: HashMap<FogIngestUri, Result<IngestSummary, String>> =
+            HashMap::new();
+        for ingest_client in self.ingest_clients.iter() {
```

Comment:
I trust Rust to handle `Vec` mutability safely, but I'm it's worth looking into grpc-rs's generated clients to ensure there isn't some unsafe interior mutability that's being exposed due to FFI or whatever.

We ended up doing `mc_connection::SyncConnection`, to wrap the `Connection` object in a lock, so I am curious as to whether this is required here.

Reply:
I thought we did that because of enclave stuff / something to do with encryption/nonce behaving properly?

Reply:
Yeah it might have something to do with retries - e.g. a piece of code constructs a Tx and encrypts, then attempts to send it 3 times since `SyncConnection` implements retrying. Can't have anything else send something during that time or decryption will fail.

Reply:
FWIW I grepped the `grpcio` code and there were some mentions of thread-safety and thread-compatibility, so I'm cautiously optimistic there aren't any lurking thread safety issues with this approach.

Reply:
I'm fine with this as-is, I just wanted to raise the issue since I didn't know why we had `SyncConnection`, and my strong suspicion was that whatever gRPC framework we were using simply wasn't getting tagged as `Sync` by `rustc` (presumably due to a raw pointer).